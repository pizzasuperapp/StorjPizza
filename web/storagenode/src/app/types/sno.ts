// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    BandwidthUsed,
    EgressUsed,
    IngressUsed,
    Node,
    SatelliteInfo,
    SatelliteScores,
    Stamp,
    Utilization,
} from '@/storagenode/sno/sno';

/**
 * Holds all node module state.
 */
export class StorageNodeState {
    public info: Node = new Node();
    public utilization: Utilization = new Utilization();
    public satellites: SatelliteInfo[] = [];
    public disqualifiedSatellites: SatelliteInfo[] = [];
    public suspendedSatellites: SatelliteInfo[] = [];
    public selectedSatellite: SatelliteInfo = new SatelliteInfo();
    public bandwidthChartData: BandwidthUsed[] = [];
    public egressChartData: EgressUsed[] = [];
    public ingressChartData: IngressUsed[] = [];
    public storageChartData: Stamp[] = [];
    public storageSummary = 0;
    public bandwidthSummary = 0;
    public egressSummary = 0;
    public ingressSummary = 0;
    public satellitesScores: SatelliteScores[] = [];
    public audits: SatelliteScores = new SatelliteScores();
}
