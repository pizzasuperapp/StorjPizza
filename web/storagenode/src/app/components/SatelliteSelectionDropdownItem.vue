// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <button
        :name="`Choose ${satellite.url} satellite`"
        class="satellite-choice"
        type="button"
        @click.stop="onSatelliteClick"
    >
        <DisqualificationIcon
            v-if="satellite.disqualified"
            class="satellite-choice__image"
            alt="disqualified image"
        />
        <SuspensionIcon
            v-if="satellite.suspended && !satellite.disqualified"
            class="satellite-choice__image"
            alt="suspended image"
        />
        <p
            class="satellite-choice__name"
            :class="{
                disqualified: satellite.disqualified,
                suspended: satellite.suspended,
                'with-id-button': isNameShown,
                'with-copy-button': !isNameShown
            }"
        >
            {{ isNameShown ? satellite.url : satellite.id }}
        </p>
        <div class="satellite-choice__right-area">
            <button
                v-if="isNameShown"
                name="Show Satellite ID"
                class="satellite-choice__right-area__button"
                type="button"
                @click.stop.prevent="toggleSatelliteView"
            >
                <EyeIcon />
                <p class="satellite-choice__right-area__button__text">ID</p>
            </button>
            <div v-else class="row">
                <button
                    v-clipboard:copy="satellite.id"
                    name="Copy Satellite ID"
                    class="satellite-choice__right-area__button copy-button"
                    type="button"
                    @click.stop="() => {}"
                >
                    <CopyIcon />
                </button>
                <button
                    name="Show Satellite Name"
                    class="satellite-choice__right-area__button"
                    type="button"
                    @click.stop.prevent="toggleSatelliteView"
                >
                    <EyeIcon />
                    <p class="satellite-choice__right-area__button__text">Name</p>
                </button>
            </div>
        </div>
    </button>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import CopyIcon from '@/../static/images/Copy.svg';
import DisqualificationIcon from '@/../static/images/disqualify.svg';
import EyeIcon from '@/../static/images/Eye.svg';
import SuspensionIcon from '@/../static/images/suspend.svg';

import { SatelliteInfo } from '@/storagenode/sno/sno';

// @vue/component
@Component({
    components: {
        DisqualificationIcon,
        SuspensionIcon,
        CopyIcon,
        EyeIcon,
    },
})
export default class SatelliteSelectionDropdownItem extends Vue {
    @Prop({default: () => new SatelliteInfo()})
    public readonly satellite: SatelliteInfo;

    /**
     * Indicates if name or id should be shown.
     */
    public isNameShown = true;

    /**
     * Toggles between name and id view.
     */
    public toggleSatelliteView(): void {
        this.isNameShown = !this.isNameShown;
    }

    /**
     * Emits action that selects satellite and fetches related information.
     */
    public onSatelliteClick(): void {
        this.$emit('onSatelliteClick', this.satellite.id);
    }
}
</script>

<style scoped lang="scss">
    .satellite-choice {
        position: relative;
        display: flex;
        width: calc(100% - 36px);
        align-items: center;
        justify-content: space-between;
        margin-left: 8px;
        border-radius: 12px;
        padding: 10px;

        &__image {
            position: absolute;
            top: 50%;
            left: 10px;
            transform: translateY(-50%);
        }

        &__name {
            font-size: 14px;
            line-height: 21px;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
        }

        &:hover {
            background-color: var(--satellite-selection-hover-background-color);
            cursor: pointer;
            color: var(--regular-text-color);
        }

        &__right-area {

            &__button {
                display: flex;
                align-items: center;
                justify-content: center;
                background: var(--button-background-color);
                border-radius: 5px;
                height: 30px;
                padding: 0 10px;
                cursor: pointer;
                font-family: 'font_medium', sans-serif;
                font-size: 13px;
                color: #9cabbe;
                border: transparent;

                &__text {
                    margin-left: 6.75px;
                }

                &:hover {
                    background-color: #e4ebfc;
                    cursor: pointer;
                    color: #133e9c;

                    .svg ::v-deep path {
                        fill: #133e9c !important;
                    }
                }
            }
        }
    }

    .disqualified,
    .suspended {
        margin-left: 24px;
    }

    .copy-button {
        margin-right: 8px;
    }

    .row {
        display: flex;
    }

    .with-id-button {
        width: calc(100% - 90px);
    }

    .with-copy-button {
        width: calc(100% - 140px);
    }
</style>
