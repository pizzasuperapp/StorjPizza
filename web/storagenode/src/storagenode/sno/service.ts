// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { StorageNodeApi } from '@/storagenode/api/storagenode';
import { Dashboard, Satellite, Satellites } from '@/storagenode/sno/sno';

/**
 * SNOService is used to store and handle node information.
 * SNOService exposes a business logic related to node.
 */
export class StorageNodeService {
    private readonly node: StorageNodeApi;

    public constructor(api: StorageNodeApi) {
        this.node = api;
    }

    /**
     * Gets dashboard data from server.
     */
    public async dashboard(): Promise<Dashboard> {
        return await this.node.dashboard();
    }

    /**
     * Gets satellite data from server.
     * @param id - satellite id
     */
    public async satellite(id: string): Promise<Satellite> {
        return await this.node.satellite(id);
    }

    /**
     * Gets all satellites data from server.
     */
    public async satellites(): Promise<Satellites> {
        return await this.node.satellites();
    }
}
