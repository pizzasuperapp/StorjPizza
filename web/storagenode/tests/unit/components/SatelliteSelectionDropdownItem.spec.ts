// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import VueClipboard from 'vue-clipboard2';

import SatelliteSelectionDropdownItem from '@/app/components/SatelliteSelectionDropdownItem.vue';

import { SatelliteInfo } from '@/storagenode/sno/sno';
import { createLocalVue, mount, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(VueClipboard);

describe('SatelliteSelectionDropdownItem', (): void => {
    it('renders correctly', (): void => {
        const wrapper = mount(SatelliteSelectionDropdownItem, {
            propsData: {
                satellite: new SatelliteInfo(
                    '1',
                    'name',
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if disqualified', (): void => {
        const wrapper = mount(SatelliteSelectionDropdownItem, {
            propsData: {
                satellite: new SatelliteInfo(
                    '1',
                    'name',
                    new Date(),
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if suspended', (): void => {
        const wrapper = mount(SatelliteSelectionDropdownItem, {
            propsData: {
                satellite: new SatelliteInfo(
                    '1',
                    'name',
                    null,
                    new Date(),
                ),
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if suspended', async (): Promise<void> => {
        const satellite = new SatelliteInfo(
            '1',
            'name',
        );

        const wrapper = shallowMount(SatelliteSelectionDropdownItem, {
            localVue,
            propsData: {
                satellite,
            },
        });

        await wrapper.find('.satellite-choice__right-area__button').trigger('click');

        expect(wrapper.find('.satellite-choice__name').text()).toBe(satellite.id);
        expect(wrapper.findAll('.satellite-choice__right-area__button').length).toBe(2);

        await wrapper.findAll('.satellite-choice__right-area__button').at(1).trigger('click');

        expect(wrapper.find('.satellite-choice__name').text()).toBe(satellite.url);
    });
});
