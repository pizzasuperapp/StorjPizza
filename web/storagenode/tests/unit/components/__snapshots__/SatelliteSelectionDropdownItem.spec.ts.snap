// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SatelliteSelectionDropdownItem renders correctly 1`] = `
<button name="Choose name satellite" type="button" class="satellite-choice">
  <!---->
  <!---->
  <p class="satellite-choice__name with-id-button">
    name
  </p>
  <div class="satellite-choice__right-area"><button name="Show Satellite ID" type="button" class="satellite-choice__right-area__button"><svg></svg>
      <p class="satellite-choice__right-area__button__text">ID</p>
    </button></div>
</button>
`;

exports[`SatelliteSelectionDropdownItem renders correctly if disqualified 1`] = `
<button name="Choose name satellite" type="button" class="satellite-choice"><svg alt="disqualified image" class="satellite-choice__image"></svg>
  <!---->
  <p class="satellite-choice__name disqualified with-id-button">
    name
  </p>
  <div class="satellite-choice__right-area"><button name="Show Satellite ID" type="button" class="satellite-choice__right-area__button"><svg></svg>
      <p class="satellite-choice__right-area__button__text">ID</p>
    </button></div>
</button>
`;

exports[`SatelliteSelectionDropdownItem renders correctly if suspended 1`] = `
<button name="Choose name satellite" type="button" class="satellite-choice">
  <!----> <svg alt="suspended image" class="satellite-choice__image"></svg>
  <p class="satellite-choice__name suspended with-id-button">
    name
  </p>
  <div class="satellite-choice__right-area"><button name="Show Satellite ID" type="button" class="satellite-choice__right-area__button"><svg></svg>
      <p class="satellite-choice__right-area__button__text">ID</p>
    </button></div>
</button>
`;
